arachne/client/arachne/generators.py

```diff
@@ -0,0 +1,373 @@
+    p : float in [0, 1]
+        probability of edge formation
+
+    create_using : Union[ar.Graph,ar.DiGraph,ar.PropGraph]
+        Arachne graph constructor
+        constructors supported
+        ar.Graph
+        ar.DiGraph
+        ar.PropGraph
+
+    Returns
+    -------
+    graph: Union[ar.Graph,ar.DiGraph,ar.PropGraph]
+    
+    """
+    m = np.random.binomial(n * (n - 1) // 2, p)  # determine number of edges
+    V, U = ak.randint(0, n, m), ak.randint(0, n, m)  # random pairs of nodes
+    graph = empty_graph(create_using)
+    if V.size == 0 or U.size == 0:
+        return graph
+    graph.add_edges_from(V,U)
+    return graph
+
+
+def karate_club_graph() -> Tuple[ak.pdarray]:
+   
+    """
+    Return's Zachary's Karate Club Graph
+
+    Each node of the 34 nodes is labelled with a ground truth community.
+
+    Returns
+    -------
+    C : ak.pdarray[int64] (34 elements)
+        community label
+    V : ak.pdarray[int64] (156 elements)
+        out nodes
+    U : ak.pdarray[int64] (156 elements)
+        in nodes
+
+    References
+    ----------
+    An Information Flow Model for Conflict and Fission in Small Groups.
+        Wayne W. Zachary. Journal of Anthropological Research, 33, 452-473
+        (1977)
+    """
+    C = ak.array([ 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 1, 1, 1, 1, 0, 0, 1, 1, 0, 1,
+                   0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0 ])
+    V = ak.array([ 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1,
+                   1, 1, 1, 1, 1, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 3, 3, 3, 3, 3,
+                   3, 4, 4, 4, 5, 5, 5, 5, 6, 6, 6, 6, 7, 7, 7, 7, 8, 8, 8, 8,
+                   8, 9, 9, 10, 10, 10, 11, 12, 12, 13, 13, 13, 13, 13, 14, 14,
+                   15, 15, 16, 16, 17, 17, 18, 18, 19, 19, 19, 20, 20, 21, 21,
+                   22, 22, 23, 23, 23, 23, 23, 24, 24, 24, 25, 25, 25, 26, 26,
+                   27, 27, 27, 27, 28, 28, 28, 29, 29, 29, 29, 30, 30, 30, 30,
+                   31, 31, 31, 31, 31, 31, 32, 32, 32, 32, 32, 32, 32, 32, 32,
+                   32, 32, 32, 33, 33, 33, 33, 33, 33, 33, 33, 33, 33, 33, 33,
+                   33, 33, 33, 33, 33 ])
+    U = ak.array([ 1, 2, 3, 4, 5, 6, 7, 8, 10, 11, 12, 13, 17, 19, 21, 31, 0, 2,
+                   3, 7, 13, 17, 19, 21, 30, 0, 1, 3, 7, 8, 9, 13, 27, 28, 32,
+                   0, 1, 2, 7, 12, 13, 0, 6, 10, 0, 6, 10, 16, 0, 4, 5, 16, 0,
+                   1, 2, 3, 0, 2, 30, 32, 33, 2, 33, 0, 4, 5, 0, 0, 3, 0, 1, 2,
+                   3, 33, 32, 33, 32, 33, 5, 6, 0, 1, 32, 33, 0, 1, 33, 32, 33,
+                   0, 1, 32, 33, 25, 27, 29, 32, 33, 25, 27, 31, 23, 24, 31, 29,
+                   33, 2, 23, 24, 33, 2, 31, 33, 23, 26, 32, 33, 1, 8, 32, 33,
+                   0, 24, 25, 28, 32, 33, 2, 8, 14, 15, 18, 20, 22, 23, 29, 30,
+                   31, 33, 8, 9, 13, 14, 15, 18, 19, 20, 22, 23, 26, 27, 28, 29,
+                   30, 31, 32 ])
+    return (V, U)  #TODO: when community detection is implemented (C,U,V)
```

Comment:
We want the karate club graph to also accepted `create_using`, create an empty graph, and then return a graph, directed graph, or prop graph.